platforms/jvm/testing-jvm/src/integTest/groovy/org/gradle/testing/testsuites/TestSuiteAutoReportingIntegrationTest.groovy

```diff
@@ -0,0 +1,208 @@
+
+        then:
+        result.assertTasksExecutedInOrder(":test", ":test2", ":generateAggregateReport")
+
+        when:
+        succeeds("help")
+
+        then:
+        result.assertTasksExecuted(":help")
+        result.assertTaskNotExecuted(":generateAggregateReport")
+    }
+
+    def "test task reports are automatically generated for jvm test suites"() {
+        given:
+        buildFile << """
+            plugins {
+                id("jvm-test-suite")
+            }
+
+            ${mavenCentralRepository()}
+
+            testing.suites.create("test", JvmTestSuite) {
+                useJUnitJupiter()
+            }
+            testing.suites.create("test2", JvmTestSuite) {
+                useJUnitJupiter()
+            }
+        """
+
+        when:
+        succeeds("test")
+
+        then:
+        result.assertTaskOrder(":test", ":generateAggregateReport")
+        aggregateResults.totalNumberOfTestClassesExecuted == 1
+        aggregateResults.testClass('ExampleTest').assertTestsExecuted('test')
+        aggregateResults.testClass('ExampleTest').testCount == 1
+
+        when:
+        succeeds("test2")
+
+        then:
+        result.assertTaskOrder(":test2", ":generateAggregateReport")
+        aggregateResults.totalNumberOfTestClassesExecuted == 1
+        aggregateResults.testClass('ExampleTest2').assertTestsExecuted('test2')
+        aggregateResults.testClass('ExampleTest2').testCount == 1
+
+        when:
+        succeeds("test", "test2")
+
+        then:
+        result.assertTaskOrder(":test", ":test2", ":generateAggregateReport")
+        aggregateResults.totalNumberOfTestClassesExecuted == 2
+        aggregateResults.testClass('ExampleTest').assertTestsExecuted('test')
+        aggregateResults.testClass('ExampleTest').testCount == 1
+        aggregateResults.testClass('ExampleTest2').assertTestsExecuted('test2')
+        aggregateResults.testClass('ExampleTest2').testCount == 1
+
+        when:
+        succeeds("help")
+
+        then:
+        result.assertTasksExecuted(":help")
+        result.assertTaskNotExecuted(":generateAggregateReport")
+    }
+
+    def "configuring tasks does not cause test aggregation task to execute"() {
+        given:
+        buildFile << """
+            plugins {
+                id("jvm-test-suite")
+            }
+
+            ${mavenCentralRepository()}
+
+            testing.suites.create("test", JvmTestSuite) {
+                useJUnitJupiter()
+                targets.test.testTask.get()
+            }
+            testing.suites.create("test2", JvmTestSuite) {
+                useJUnitJupiter()
+                targets.test2.testTask.get()
+            }
+
+            tasks.named("generateAggregateReport").configure {
+                throw new RuntimeException("Should not be executed")
+            }
+        """
+
+        expect:
+        succeeds("help")
+    }
+
+    def "can run aggregate report without requesting tests as well"() {
```

Comment:
💭 I wonder if this is the result we want here.  Maybe this should be some sort of non-user-invocable task?